spec/root_spec.rb

```diff
@@ -0,0 +1,315 @@
+          .to eq('seriousKitten' => { 'weight'  => '3 kilos',
+                                      'catFood' => { 'amount' => 3, 'brand' => 'Whiskas' } })
+      end
+
+      it 'wraps include_root' do
+        expect(JSON.parse(instance.surrealize(root: 'serious_kitten', include_root: true)))
+          .to eq('serious_kitten' => { 'serious_cat' => { 'weight'   => '3 kilos',
+                                                          'cat_food' => { 'amount' => 3,
+                                                                          'brand' => 'Whiskas' } } })
+      end
+
+      it 'wraps include_namespaces' do
+        expect(JSON.parse(instance.surrealize(root: 'serious_kitten', include_namespaces: true)))
+          .to eq('serious_kitten' => { 'serious_cat' => { 'weight'   => '3 kilos',
+                                                          'cat_food' => { 'amount' => 3,
+                                                                          'brand' => 'Whiskas' } } })
+      end
+    end
+
+    context 'with nested classes' do
+      let(:instance) { Animal::Dog.new }
+
+      it 'builds schema' do
+        expect(instance.build_schema(root: 'dog')).to eq(dog: { breed: 'Collie' })
+      end
+
+      it 'surrealizes' do
+        expect(JSON.parse(instance.surrealize(root: 'dog')))
+          .to eq('dog' => { 'breed' => 'Collie' })
+      end
+
+      it 'camelizes' do
+        expect(instance.build_schema(root: 'dog', camelize: true))
+          .to eq(dog: { breed: 'Collie' })
+
+        expect(JSON.parse(instance.surrealize(root: 'dog', camelize: true)))
+          .to eq('dog' => { 'breed' => 'Collie' })
+      end
+
+      it 'wraps include_root' do
+        expect(JSON.parse(instance.surrealize(root: 'dog', include_root: true)))
+          .to eq('dog' => { 'dog' => { 'breed' => 'Collie' } })
+      end
+
+      it 'wraps include_namespaces' do
+        expect(JSON.parse(instance.surrealize(root: 'dog', include_namespaces: true)))
+          .to eq('dog' => { 'animal' => { 'dog' => { 'breed' => 'Collie' } } })
+      end
+    end
+
+    context 'Module::Class' do
+      let(:instance) { Instrument::Guitar.new }
+
+      it 'builds schema' do
+        expect(instance.build_schema(root: 'guitar'))
+          .to eq(guitar: { brand_name: 'Fender' })
+      end
+
+      it 'surrealizes' do
+        expect(JSON.parse(instance.surrealize(root: 'guitar')))
+          .to eq('guitar' => { 'brand_name' => 'Fender' })
+      end
+
+      it 'camelizes' do
+        expect(instance.build_schema(root: 'guitar', camelize: true))
+          .to eq(guitar: { brandName: 'Fender' })
+
+        expect(JSON.parse(instance.surrealize(root: 'guitar', camelize: true)))
+          .to eq('guitar' => { 'brandName' => 'Fender' })
+      end
+
+      it 'include_root' do
+        expect(JSON.parse(instance.surrealize(root: 'guitar', include_root: true)))
+          .to eq('guitar' => { 'guitar' => { 'brand_name' => 'Fender' } })
```

Comment:
Actually, I find this really confusing. If I wrote `instance.surrealize(include_root: true, root: 'guitar')` I would expect only one root, e.g. `{ 'guitar' => { 'brand_name' => 'Fender' } }` in this case. But I don't know if I am right on this. @AlessandroMinali ?

Reply:
For what makes sense for me is that root param overrides include_root.

```
instance.surrealize(include_root: true, root: 'car')
# => { 'car' => { 'brand_name' => 'Fender' } }

instance.surrealize(root: 'car')
# => { 'car' => { 'brand_name' => 'Fender' } }

instance.surrealize(include_root: true)
# => { 'guitar' => { 'brand_name' => 'Fender' } }

instance.surrealize()
# => { 'brand_name' => 'Fender' }

Reply:
Exactly.

Reply:
So would `root` override `include_namespaces` too?

Reply:
@chrisatanasian yeah, I guess so.